src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -1016,16 +1031,16 @@ private function data_wp_router_region_processor( WP_Interactivity_API_Directive
 		if ( 'enter' === $mode && ! $this->has_processed_router_region ) {
 			$this->has_processed_router_region = true;
 
-			// Initialize the `core/router` store.
+			/*
+			 * Initialize the `core/router` store.
+			 * If the store is not initialized like this with minimal
+			 * navigation object, the interactivity-router script module
+			 * errors.
+			 */
 			$this->state(
 				'core/router',
 				array(
-					'navigation' => array(
-						'texts' => array(
-							'loading' => __( 'Loading page, please wait.' ),
-							'loaded'  => __( 'Page Loaded.' ),
-						),
-					),
+					'navigation' => new stdClass(),
```

Comment:
Maybe this is better?
```suggestion
					'navigation' => array(
						"hasStarted" => false,
						"hasFinished" => false,
					),
```

There's a bug where if `navigation` is not defined on the server, interactivity-router may error.

Reply:
To confim my understanding: `	'navigation' => new stdClass()` currently fixes that bug, right?

Reply:
Yes, this ensures the `navigation` is an object (`{}`). Then there's no failure when it assigns to `navigation.hasStarted` .